src/components/WorkPoolBaseJobTemplateFormSection.vue

```diff
@@ -0,0 +1,118 @@
+      <template v-else>
+        <p-message warning>
+          This work pool's base job template does not have any customizations. To add customizations,
+          edit the base job template directly with the <b>Advanced</b> tab.
+        </p-message>
+      </template>
+    </template>
+    <template #advanced>
+      <p-message info>
+        This is the JSON representation of the base job template. A work pool's based job template
+        controls infrastructure configuration for all flow runs in the work pool, and specifies the
+        configuration that can be overridden by deployments.
+        <br>
+        <br>
+        For more information on the structure of a work pool's base job template, check out
+        <p-link :to="localization.docs.workPools">
+          the docs.
+        </p-link>.
+      </p-message>
+      <JsonInput :model-value="baseJobTemplateJson" show-format-button @update:model-value="onJsonUpdate" />
+    </template>
+  </p-tabs>
+</template>
+
+<script lang="ts" setup>
+  import { isEqual, cloneDeep } from 'lodash'
+  import { computed, ref, watch } from 'vue'
+  import { SchemaFormFieldsWithValues, BetaBadge, JsonInput } from '@/components'
+  import { useTabs } from '@/compositions'
+  import { localization } from '@/localization'
+  import { mapper } from '@/services'
+  import { Schema, SchemaProperty, SchemaValues, WorkerBaseJobTemplate } from '@/types'
+  import { getSchemaDefaults, getSchemaWithoutDefaults } from '@/utilities'
+
+
+  const props = defineProps<{
+    baseJobTemplate: WorkerBaseJobTemplate,
+  }>()
+
+  const emit = defineEmits<{
+    (e: 'update:base-job-template', json: WorkerBaseJobTemplate): void,
+  }>()
+
+  const computedTabs = computed(() => [
+    { label: 'Defaults' },
+    { label: 'Advanced' },
+  ])
+
+  const { tab, tabs } = useTabs(computedTabs)
+
+  const onJsonUpdate = (json: string): void => {
+    localBaseJobTemplateJson.value = json
+    try {
+      emit('update:base-job-template', JSON.parse(json))
+    } catch (ex) {
+      // don't emit if the local value is not valid JSON
+    }
+  }
+
+  const onDefaultValuesUpdate = (values: SchemaValues): void => {
+    const { baseJobTemplate } = props
+    const keys = Object.keys(baseJobTemplate.variables?.properties ?? {})
+
+    const newTemplate = {
+      ...baseJobTemplate,
+      variables: {
+        ...baseJobTemplate.variables,
+        properties: keys.reduce<Record<string, SchemaProperty>>((acc, key) => {
+          acc[key] = {
+            ...baseJobTemplate.variables?.properties?.[key],
+            default: values[key],
+          }
+          return acc
+        }, {}),
+      },
+    }
+
+    emit('update:base-job-template', newTemplate)
+  }
+
+  const localBaseJobTemplateJson = ref<string | null>(null)
+  const baseJobTemplateJson = computed<string>(() => localBaseJobTemplateJson.value ?? JSON.stringify(props.baseJobTemplate, null, 2))
+  watch(() => props.baseJobTemplate, (prev, current) => {
```

Comment:
I'm trying to determine when the parent has updated the template that should be rendered. This happens when the selected type of worker changes. I've simplified the logic a little bit by removing a computed property, but I think it might be better if I add another prop to communicate when the type has changed.

Reply:
What still doesn't make sense is your checking if they're the same object and then assigning the local variable equal to the prop. But they're the same right? 

If the isEqual check returns true isn't `localBaseJobTemplateJson` already the same as `stringify(current)`? 

```
if (isEqual(JSON.parse(localBaseJobTemplateJson.value), cloneDeep(current))) {
  localBaseJobTemplateJson.value = stringify(current)
}
```